packages/flutter/test/material/tabs_test.dart

```diff
@@ -386,6 +386,37 @@ void main() {
       material3 ? const Size(14.25, 72.0) : const Size(14.0, 72.0));
   });
 
+  testWidgets('Tab icon-label margin', (WidgetTester tester) async {
+    Widget buildApp({required bool useMaterial3}) {
+      final ThemeData theme = ThemeData(fontFamily: 'FlutterTest', useMaterial3: useMaterial3);
+      return MaterialApp(theme: theme, home: const Center(child: Material(
+        child: Tab(
+          icon: Icon(Icons.house),
+          text: 'x',
+        )
+      )));
+    }
+
+    double getMarginBetweenIconAndLabel() {
+      final Rect iconRect = tester.getRect(find.byIcon(Icons.house));
+      final Rect labelRect = tester.getRect(find.text('x'));
+      return labelRect.top - iconRect.bottom;
+    }
+
+    // Material 2
+    await tester.pumpWidget(
+      buildApp(useMaterial3: false),
+    );
+    expect(getMarginBetweenIconAndLabel(), 10);
+
+    // Material 3
+    await tester.pumpWidget(
+      buildApp(useMaterial3: true),
+    );
```

Comment:
Material 2 and Material 3 require separate tests. Please search for other similar in the framework for reference.

Reply:
Add a Separate Material 3 similar to one below, notice the title. (You can skip passing ThemeData in this test)

https://github.com/flutter/flutter/blob/5f17badf4118b474f0b97ee1afe0773ac92b1e4b/packages/flutter/test/material/filter_chip_test.dart#L239-L241

And a separate Material 2 test similar to one blow (For Material 2, set `useMaterial3` to `false` and pass a ThemeData in the `MaterialApp.theme` parameter directly)

https://github.com/flutter/flutter/blob/5f17badf4118b474f0b97ee1afe0773ac92b1e4b/packages/flutter/test/material/filter_chip_test.dart#L134-L140